torchvision/datasets/folder.py

```diff
@@ -31,16 +32,22 @@ def is_image_file(filename):
     return has_file_allowed_extension(filename, IMG_EXTENSIONS)
 
 
-def make_dataset(directory, class_to_idx, extensions=None, is_valid_file=None):
+def make_dataset(
+    directory: str,
+    class_to_idx: Dict[str, int],
+    extensions: Optional[Tuple[str, ...]] = None,
+    is_valid_file: Optional[Callable[[str], bool]] = None,
```

Comment:
Hum, from https://stackoverflow.com/questions/55231989/optional-union-in-type-hint this should not be possible?

Reply:
Ok, I missed one set of brackets while reading the code